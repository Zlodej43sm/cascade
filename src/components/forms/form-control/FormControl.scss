@use '../../../styles/colors';
@use '../../../styles/typography';
@use '../../../styles/sizes';


.wrapper {
  display: flex;
  flex-direction: column;
}

.label {
  @include typography.poppins(17px, 25px, 600);
  display: block;
  color: colors.$cascade-primary;
  margin-bottom: sizes.$base;
}
