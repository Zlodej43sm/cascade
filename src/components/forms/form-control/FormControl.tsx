/* eslint-disable jsx-a11y/label-has-associated-control */
import React from 'react';

import styles from './FormControl.scss';

export type FormControlProps = {
  id?: string
  label?: string
  children?: React.ReactNode
};

function FormControl({ label, children }: FormControlProps) {
  return label ? (
    <label className={styles.wrapper}>
      <span className={styles.label}>{label}</span>
      {children}
    </label>
  ) : (
    <>{children}</>
  );
}

export default FormControl;
