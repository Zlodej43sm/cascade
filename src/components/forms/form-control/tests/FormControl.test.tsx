import React from 'react';
import { render } from '@testing-library/react';

import FormControl from '../FormControl';

describe('FormControl', () => {
  it('Should match snapshot', () => {
    const { asFragment } = render(<FormControl label="Label">children</FormControl>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('Should match snapshot without label', () => {
    const { asFragment } = render(<FormControl>children</FormControl>);
    expect(asFragment()).toMatchSnapshot();
  });
});
