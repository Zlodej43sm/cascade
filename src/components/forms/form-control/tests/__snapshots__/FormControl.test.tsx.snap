// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormControl Should match snapshot 1`] = `
<DocumentFragment>
  <label
    class="wrapper"
  >
    <span
      class="label"
    >
      Label
    </span>
    children
  </label>
</DocumentFragment>
`;

exports[`FormControl Should match snapshot without label 1`] = `
<DocumentFragment>
  children
</DocumentFragment>
`;
